src/lib/spriteActor.js

```diff
@@ -0,0 +1,50 @@
+async function loadImage(src) {
+    let img = new Image();
+    img.src = src;
+    return new Promise((res, rej) => {
+        img.onload = () => {
+            res(img);
+        }
+    });
+}
+
+export default class SpriteActor extends Actor{
+    constructor(bounds, spriteObj){
+        super(bounds);
+        this.properties = spriteObj.properties || {};
+        this.states = spriteObj.states || {idle: {location: null, rps: 5, loop: true, frames: [{x: 0, y: 0, width: 0, height: 0}]}};
+
+    }
+
+    loadSpriteObj = (spriteObj) => {
+        this.properties = spriteObj.properties;
+        this.states = spriteObj.states;
+
+        //what is diffing images?
+    }
+
+    create = () => {
+        (async () => {
+            await Promise.all(Object.keys(sprites.states).map(async (s) => {
+                let img = await loadImage(sprites.states[s].location);
+                Object.assign(sprites.states[s], { image: img });
+            }));
+            console.log(sprites);
+        })()
+    }
+
+    render = (deltaTime) => {
+        if(this.states[0].loop == true){
+            let numFrames = this.states[0].frames.length;
```

Comment:
Add the check for RPS(Render per second) 
1. Have a counter keep track of how much time has gone by
2. Check if that is greater than the state RPS

If it is -> switch frames